recruit_flow_ai/ai.py

```diff
@@ -0,0 +1,218 @@
+"""
+This module contains the RecruitFlowAI class, which is an AI assistant for the 
+RecruitFlow application.
+
+The RecruitFlowAI class provides methods for generating responses based on given 
+messages using the OpenAI API.
+
+Example:
+    # Create an instance of RecruitFlowAI
+    ai = RecruitFlowAI(api_key="YOUR_API_KEY")
+
+    # Generate a response
+    response = ai.generate_response(openai_msgs=[
+        {"role": "system", "content": "You are a recruitment assistant."},
+        {"role": "user", "content": "What questions should I ask during the screening process"}
+    ])
+
+    # Print the response
+    logger.error(response)
+
+"""
+
+import openai
+
+import logging
+import re   # used for api key format validation
+import os   # used to read OPANAI_API_KEY env variable
+from dotenv import load_dotenv
+from pathlib import Path
+
+from .parse_config import parse_config, ConfigModel
+
+PROMPTS_CONFIG_PATH = "recruit_flow_ai/prompts_config.json"
+ISSUES_REPORT_MSG = "Report this to #recruitflowai_issues channel."
+
+logFormatter = logging.Formatter("%(asctime)s [%(threadName)-12.12s] [%(levelname)-5.5s]  %(message)s")
+logger = logging.getLogger(__name__)
+logger.setLevel("DEBUG")
+
+consoleHandler = logging.StreamHandler()
+consoleHandler.setFormatter(logFormatter)
+logger.addHandler(consoleHandler)
+
+class RecruitFlowAI:
+    """
+    AI assistant for the RecruitFlow application.
+
+    Attributes:
+        model (str): The name of the GPT model to use.
+        api_key (str): The API key for accessing the OpenAI API.
+        system_prompt (str): The system prompt to use for generating responses.
+        temperature (float): The temperature parameter for controlling the randomness of the generated responses.
+
+    Methods:
+        __init__(api_key=""): Initializes the RecruitFlowAI object with the provided API key.
+        generate_response(openai_msgs: []): Generates a response based on the given messages.
+        is_valid_api_key_format(api_key: str): Validates the format of an OpenAI API key.
+
+    """
+    model = "gpt-3.5-turbo"
+    api_key = ""
+    system_prompt = "I am helpful recruitment assistant."
+    temperature = 0
+
+    def __init__(self, api_key = ""):
+        """
+        Initializes the RecruitFlowAI object with the provided API key.
+
+        Args:
+            api_key (str): The API key for accessing the OpenAI API. If not provided, 
+            it will be read from the .env file.
+
+        Raises:
+            ValueError: If the API key is not provided or if it is not valid.
+        """
+        if api_key == "":
```

Comment:
Lets move api key extraction and validation to `settings.py` like we did before. Check other modules and https://docs.pydantic.dev/latest/usage/pydantic_settings/

Reply:
Ok, I will look into it.

Reply:
You can specify the path of the env file in class definition, it will work.

Reply:
when you add env prefix the value shouldn't contain it. 
```
class Settings():
api_key: SecretStr
```

Reply:
done [aa2f31c](https://github.com/COXIT-CO/RecruitFlowAI/pull/20/commits/aa2f31c991505ff3579a471e9abc27cca7ac8a3b)